es6-shim/es6-shim.d.ts

```diff
@@ -466,7 +466,7 @@ interface PromiseLike<T> {
     * @returns A Promise for the completion of which ever callback is executed.
     */
     then<TResult>(onfulfilled?: (value: T) => TResult | PromiseLike<TResult>, onrejected?: (reason: any) => TResult | PromiseLike<TResult>): PromiseLike<TResult>;
-    then<TResult>(onfulfilled?: (value: T) => TResult | PromiseLike<TResult>, onrejected?: (reason: any) => void): PromiseLike<TResult>;
+    then<TResult>(onfulfilled?: (value: T) => TResult | PromiseLike<TResult>, onrejected?: (reason: any) => any): PromiseLike<TResult>;
```

Comment:
I disagree with this case. The reason `void` was used here was to catch this case:

```ts
numberPromise.then(v => v + 1, e => { throw new Error("Some more detailed message: " + e.message); }); // Promise<number>
```

This is because any function that only throws is inferred as `void`.